CoreUtilities/src/au/gov/asd/tac/constellation/utilities/text/SpellChecker.java

```diff
@@ -0,0 +1,305 @@
+ */
+public final class SpellChecker {
+    private final SpellCheckingTextArea textArea;
+    private static final List<String> misspells = new ArrayList<>();
+    private List<Object> matches = new ArrayList<>();
+    private int indexOfMisspelledTextUnderCursor;       // position of the current misspelled text in misspells list
+    private final ListView<String> suggestions = new ListView<>(FXCollections.observableArrayList());
+    private Object langTool;
+    private static Object langToolStatic;
+    private Object spellingCheckRule;
+    private Popup popup = new Popup();
+    private final Label labelMessage = new Label();
+    private boolean turnOffSpellChecking = false;
+    private int startOfMisspelledTextUnderCursor;
+    private int endOfMisspelledTextUnderCursor;
+    private String specificRuleId;
+    private static final Logger LOGGER = Logger.getLogger(SpellChecker.class.getName());
+    private static final double POPUP_PADDING = 5;
+    private static final double ITEM_HEIGHT = 24;
+    private static Object language = null;
+
+    protected static final CompletableFuture<Void> LANGTOOL_LOAD;
+
+    static {
+        // langToolStatic is used to initialize the JLanguageTool at the loading, because
+        // the very first initializing of JLanguageTool is slow but after that it is fast.
+        LANGTOOL_LOAD = CompletableFuture.supplyAsync(new Supplier<Void>() {
+            @Override
+            public Void get() {
+                LanguagetoolClassLoader.loadDependencies();
+                if (MultiThreadedJLanguageTool != null) {
+                    try {
+                        Constructor<?> constructor = Languages.getDeclaredConstructor();
+                        constructor.setAccessible(true);
+                        Object languages = constructor.newInstance();
+                        language = Languages.getMethod("getLanguageForShortCode", String.class).invoke(languages, "en-AU");
+                        langToolStatic = MultiThreadedJLanguageTool.getDeclaredConstructor(Language).newInstance(language);
+
+                        //perform a check here to prevent the spell checking being too slow at the first word after loading costy
+                        final Object initMatches = JLanguagetool.getMethod("check", String.class).invoke(langToolStatic, "random text");
+
+                    } catch (NoSuchMethodException | SecurityException | InstantiationException | IllegalAccessException | IllegalArgumentException | InvocationTargetException ex) {
+                        logAndDisplayErrorMessage("Error while initializing spell checking. Spell checking may not be functioning.", ex);
+                    }
+                } else {
+                    NotifyDisplayer.display("Error while loading spell checking. Spell checking will not be functioning.", NotifyDescriptor.ERROR_MESSAGE);
+                }
+                return null;
+            }
+        }, Executors.newSingleThreadExecutor());
```

Comment:
Should this be using the global thread pool? (answer could be no, but if yes, we should look to use that instead)

Reply:
Not quite sure, just did the same as other places a static initializer is used, i.e., RunPane.java and DataAccessPaneState.java.  A static initializer is executed only once when the class is first loaded into memory. I guess that's why a global thread pool is not necessary?